packages/block-library/src/categories/edit.js

```diff
@@ -46,7 +46,11 @@ export default function CategoriesEdit( {
 		'taxonomy'
 	);
 
-	const taxonomies = allTaxonomies?.filter( ( t ) => t.visibility.public );
+	const taxonomies = allTaxonomies?.filter(
+		( t ) =>
+			t.visibility.public &&
+			( taxonomySlug === 'category' ) === ( t.slug === 'category' )
```

Comment:
What is the risk if the `category is listed here? When switching a choice a different block variation would get selected but otherwise it should remain seamless for the user.

Reply:
Yeah, @fabiankaegy also suggested that [over here](https://github.com/WordPress/gutenberg/pull/65272#discussion_r1761198688). I'll quote [my reply](https://github.com/WordPress/gutenberg/pull/65272#discussion_r1761318749) here for convenience:

> TBH, that would feel inconsistent to me in terms of UX; I think it would be somewhat surprising to the user if selecting "Categories" as the taxonomy from that dropdown gets special treatment, while all other taxonomies don't 😕 
>
> ![dropdown-includes everything](https://github.com/user-attachments/assets/69b61f8d-3222-46bf-9c61-e212617108ad)

I'm open to revisiting this. It's something I'd like to get feedback from designers on, though.

Reply:
I've updated the screencast in the PR description accordingly. 

<details>
<summary>
Here's the previous one (with Categories removed from the dropdown)
</summary>

![separate-categories-variation](https://github.com/user-attachments/assets/5c096cbe-c4af-49f2-8ecf-d7ac1ee64a6d)
</details>